src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -261,6 +267,12 @@ public JobHostTraceConfiguration Tracing
         }
 
         /// <summary>
+        /// get host-level metadata which the extension may read to do configuration. 
```

Comment:
Currently get/set - also JObject is mutable. Do we want to lock the configuration down so that an extension can read entire config but only modify its own section?